modules/apps/map/map-common/src/main/resources/META-INF/resources/js/MapBase.js

```diff
@@ -62,15 +60,48 @@ const pendingCallbacks = {};
  * @abstract
  * @review
  */
-class MapBase extends State {
+class MapBase extends EventEmitter {
+	get position() {
+		return this._STATE_.position;
+	}
+
+	set position(position) {
+		this.emit('positionChange', {
+			newVal: {
+				address: position.address,
+				location: position.location,
+			},
+		});
+
+		this._STATE_.position = position;
+	}
 
 	/**
 	 * MapBase constructor
 	 * @param  {Array} args List of arguments to be sent to State constructor
 	 * @review
 	 */
-	constructor(...args) {
-		super(...args);
+	constructor(args) {
+		super(args);
+
+		const {
+			boundingBox = {},
+			controls = [],
+			geolocation = true,
+			position = {},
+			tileURI = '//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+		} = args;
+
+		this._STATE_ = {
+			position,
+		};
+
+		this.boundingBox = boundingBox;
+		this.controls = controls;
+		this.geolocation = geolocation;
+		this.position = position;
+		this.tileURI = tileURI;
```

Comment:
Why did this get moved into this class? I think it is only used in `MapOpenStreetMap.es.js`

Reply:
So keeping in mind how inheritance works, shouldn't it be reverse than what you said here @bryceosterhaus ?

As per the [example](https://javascript.info/class-inheritance) 
- `Animal` is the base of `Rabbit`
-  `Animal` has `this.speed = args.speed`

You're saying here that `Rabbit` should have `this.speed = args.speed`, instead of `Animal` having it?

The problem being that, for example, we're using `this.controls`here, but we aren't setting it in the parent `MapBase` class